metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -2704,11 +2720,15 @@ def emit_sheets_as_charts(
             chart_snapshot.aspects.append(owner)
 
         #  Tags
-        tags = self.get_tags(sheet)
-        if tags:
-            chart_snapshot.aspects.append(
-                builder.make_global_tag_aspect_with_tag_list(tags)
-            )
+        tags = self.get_tags(sheet) or []
+        if len(self.config.tags_for_hidden_assets) > 0 and not sheet.get(c.LUID):
+            # Add hidden tags if sheet is hidden (blank luid)
+            tags.extend(self.config.tags_for_hidden_assets)
+
+        chart_snapshot.aspects.append(
```

Comment:
I think there was an issue in general with the tag ingestion. If all tags were removed in Tableau they wouldn't get removed in Datahub because tags are only saved if there are any. Now, we basically always save the GlobalTags aspect even if there are no tags. But this allows us to properly reset the tags if they were removed from Tableau.

This resulted in quite a big changeset in the golden test files. Let me know if you see an issue with this. Thanks.